pysatNASA/instruments/icon_mighti.py

```diff
@@ -149,38 +110,166 @@ def init(self):
     return
 
 
-def default(inst):
-    """Default routine to be applied when loading data.  Adjusts epoch
-    timestamps to datetimes and removes variable preambles.
+def preprocess(self, keep_original_names=False):
+    """Adjusts epoch timestamps to datetimes and removes variable preambles.
 
     Parameters
-    -----------
-    inst : pysat.Instrument
-        Instrument class object
+    ----------
+    keep_original_names : boolean
+        if True then the names as given in the netCDF ICON file
+        will be used as is. If False, a preamble is removed. (default=False)
 
     """
 
-    mm_gen.convert_timestamp_to_datetime(inst, sec_mult=1.0e-3)
-    if not inst.kwargs['keep_original_names']:
-        remove_preamble(inst)
+    mm_gen.convert_timestamp_to_datetime(self, sec_mult=1.0e-3)
+    if not keep_original_names:
+        remove_preamble(self)
+    return
 
 
-def remove_preamble(inst):
-    """Removes preambles in variable names"""
+def clean(self):
+    """Provides data cleaning based upon clean_level.
 
-    target = {'los_wind_green': 'ICON_L21_',
-              'los_wind_red': 'ICON_L21_',
-              'vector_wind_green': 'ICON_L22_',
-              'vector_wind_red': 'ICON_L22_',
-              'temperature': ['ICON_L1_MIGHTI_{}_'.format(inst.sat_id.upper()),
-                              'ICON_L23_MIGHTI_{}_'.format(inst.sat_id.upper()),
-                              'ICON_L23_']}
-    mm_gen.remove_leading_text(inst, target=target[inst.tag])
+    Note
+    ----
+        Supports 'clean', 'dusty', 'none'
+
+    """
+
+    def _clean_vars(var_list, flag, min_level):
+        """Cleans parameters in a list according to standard flags
+
+        Parameters
+        ----------
+        var_list : list of strings
+            List of variables to be cleaned.  Must match variables present in
+            the data set.
+        flag : string
+            The variable name to be used as the quality flag
+        min_level : float
+            The value at or above where we are confident in the data.  For
+            MIGHTI, these are generally 0.5 or 1.0
+
+        """
+        for var in var_list:
+            self[var] = self[var].where(self[flag] >= min_level)
+        return
```

Comment:
Would this be better placed in `pysatNASA.instruments.methods.icon.py`?

Reply:
This particular routine is localized to MIGHTI.  The other instruments use it differently (`<=` in some cases), and I'm worried moving that up the chain would create more confusion.

Reply:
renamed the FUV clean parameter as `max_level` to avoid confusion